src/main/java/org/sagebionetworks/web/server/servlet/AppConfigServlet.java

```diff
@@ -95,11 +97,31 @@ public void initializeConfigSupplier() {
       );
   }
 
+  public JSONObjectAdapter getLastConfigValueOrDefault() {
+    if (lastConfigValue != null) {
+      return lastConfigValue;
+    }
+    try {
+      return new JSONObjectAdapterImpl(DEFAULT_CONFIG_VALUE);
+    } catch (JSONObjectAdapterException e) {
+      logger.log(
+        Level.SEVERE,
+        "JSONObjectAdapterException occurred in default configuration",
+        e
+      );
+      throw new RuntimeException(e);
+    }
+  }
+
   public JSONObjectAdapter getLatestConfiguration() {
     try {
       if (configurationToken == null) {
-        logger.log(Level.SEVERE, "returning default config");
-        return new JSONObjectAdapterImpl(DEFAULT_CONFIG_VALUE);
+        logger.log(
+          Level.SEVERE,
+          "The configuration token is null, the last config value will be returned." +
+          " This usually means that the initial call to initialize the configuration session failed."
+        );
+        return getLastConfigValueOrDefault();
       }
```

Comment:
It seems like this error case will only happen when the initial configuration failed, so I updated the log message to give context.